### PR TITLE
Added tests for extract

### LINKED ISSUE #88: Tests for extrat script
# Description
Write comprehensive unit tests for each function in the extract script to ensure correctness, edge case handling, and maintainability. The goal is to cover both typical and atypical input scenarios for all core functionalities.

# Outcome
Achieve at least 90% code coverage for the extract module.
Tests run automatically in the CI pipeline.

# Associated User Stories
As a developer, I want to ensure that the extract script functions correctly in all scenarios, so that I can refactor or extend it with confidence without introducing regressions.

### PR DESCRIPTION
**Pull request by b-yacquub**
# Description
Added a test file to test each function in the extract script. Also abstracted the extract script a bit more so only one function handle making requests to the website. This was done to have purer functions and make them easier to test

Closes: #88 

# Type of change
- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

# Mentions
@Tristenx @Zephvv @yvonne @b-yacquub @nikki-w

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,3 +13,4 @@ package-lock.json
 *.tfstate.backup
 *.tfstate*
 *.tfvars.json
+.coverage
```
Comment (Tristenx) [inline .gitignore:16]: Good addition!

File: scraper/steam/extract.py [hunk 2]
```diff
@@ -44,32 +44,38 @@ def get_products(conn: connection) -> list[dict]:
     return products
 
 
-def is_discounted(url: str, discounted_class: str, headers: dict[str:str]) -> bool:
-    """Checks if the product price_class is present on the webpage."""
+def get_html_text(url: str, headers: dict[str:str]):
+    '''Gets the full text response of the html'''
```
Comment (Tristenx) [inline scraper/steam/extract.py:48]: Could you change ''' to """ so its consistent with the rest of the file.

File: scraper/steam/extract.py [hunk 3]
```diff
@@ -44,32 +44,38 @@ def get_products(conn: connection) -> list[dict]:
...
-def scrape_price(url: str, cost_class: str, headers: dict[str:str]) -> str:
+def is_discounted(html: str, discounted_class: str) -> bool:
+    """Checks if the product price_class is present on the webpage."""
+    soup = BeautifulSoup(html[1], "html.parser")
+    if soup.find(attrs={"class": discounted_class}) is not None:
+        return True
+    return False
+
+
+def scrape_price(html: str, cost_class: str) -> str:
```
Comment (Tristenx) [inline scraper/steam/extract.py:63]: I like that this is a pure function now.

File: scraper/steam/extract.py [hunk 4]
```diff
@@ -44,32 +44,38 @@ def get_products(conn: connection) -> list[dict]:
     return products
 
 
-def is_discounted(url: str, discounted_class: str, headers: dict[str:str]) -> bool:
-    """Checks if the product price_class is present on the webpage."""
+def get_html_text(url: str, headers: dict[str:str]):
```
Comment (Tristenx) [inline scraper/steam/extract.py:47]: Type hint please

File: scraper/steam/extract.py [hunk 5]
```diff
@@ -94,6 +100,9 @@ def add_price_to_products(products: dict[str:str], cost_class: str, discounted_c
     steam_cost_class = "game_purchase_price price"
     steam_discounted_class = "discount_final_price"
 
+    print(get_current_price('https://store.steampowered.com/app/990080/Hogwarts_Legacy/', steam_cost_class,
```
Comment (Tristenx) [inline scraper/steam/extract.py:103]: Remove the rouge print statement please

**Review by Tristenx — CHANGES_REQUESTED** [review]
Good tests and some great changes to extract.py. Just a few small tweaks please.

File: scraper/steam/extract.py (hunk 5 above)
Comment (yvonne0711) [inline scraper/steam/extract.py:103]: Do we want to print this out or is this for testing purposes?

**Review by yvonne0711 — COMMENTED** [review]
Made a few comments but other than that, the tests look really good, it will also be reusable for the other websites.

File: scraper/steam/extract.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (b-yacquub) [reply scraper/steam/extract.py]: Yh it makes it so only one function is making a request

**Review by Tristenx — DISMISSED** [review]
Thanks for making those changes

**Review by Tristenx — APPROVED** [review]
Nice fix on conflicts!
